server/queues/processors/NotificationsProcessor.ts

```diff
@@ -71,66 +98,82 @@ export default class NotificationsProcessor extends BaseProcessor {
         },
       ],
     });
-    const eventName =
-      event.name === "documents.publish" ? "published" : "updated";
 
-    for (const setting of notificationSettings) {
-      // Suppress notifications for suspended users
-      if (setting.user.isSuspended) {
-        continue;
-      }
+    for (const recipient of recipients) {
+      const notify = await shouldNotify({
+        user: recipient.user,
+        userId: recipient.userId,
+        document: document,
+        documentId: event.documentId,
+        event: event,
+        eventName: "published",
+      });
 
-      // For document updates we only want to send notifications if
-      // the document has been edited by the user with this notification setting
-      // This could be replaced with ability to "follow" in the future
-      if (
-        eventName === "updated" &&
-        !document.collaboratorIds.includes(setting.userId)
-      ) {
-        continue;
+      if (notify) {
+        await DocumentNotificationEmail.schedule({
+          to: recipient.user.email,
+          eventName: "published",
+          documentId: document.id,
+          teamUrl: team.url,
+          actorName: document.updatedBy.name,
+          collectionName: collection.name,
+          unsubscribeUrl: recipient.unsubscribeUrl,
+        });
       }
+    }
+  }
 
-      // Check the user has access to the collection this document is in. Just
-      // because they were a collaborator once doesn't mean they still are.
-      const collectionIds = await setting.user.collectionIds();
+  async revisionCreated(event: RevisionEvent) {
+    const [collection, document, team] = await Promise.all([
+      Collection.findByPk(event.collectionId),
+      Document.findByPk(event.documentId),
+      Team.findByPk(event.teamId),
+    ]);
 
-      if (!collectionIds.includes(document.collectionId)) {
-        continue;
-      }
+    if (!document || !team || !collection) {
+      return;
+    }
 
-      // If this user has viewed the document since the last update was made
-      // then we can avoid sending them a useless notification, yay.
-      const view = await View.findOne({
-        where: {
-          userId: setting.userId,
-          documentId: event.documentId,
-          updatedAt: {
-            [Op.gt]: document.updatedAt,
-          },
+    // Document update notifications
+    // Should be similar to notifications.
+    const subscriptions = await Subscription.findAll({
```

Comment:
We still need to consider the `NotificationSetting` here as document update emails may be disabled entirely, in which case all of the users subscriptions are essentially disabled

Reply:
Fixing this should be as simple as calling email notifier inside this `if` block https://github.com/outline/outline/blob/11b823e24266201b2c61a227d3d38c7dd4d2295e/server/queues/processors/NotificationsProcessor.ts#L291 although I am quite sure I removed it intentionally because 

https://github.com/outline/outline/blob/30190866f8c2c6189105d34cb6b508f4917b4564/server/queues/processors/NotificationsProcessor.ts#L31 (on **main**) handles both `published` and `updated` events

https://github.com/outline/outline/blob/30190866f8c2c6189105d34cb6b508f4917b4564/server/queues/processors/NotificationsProcessor.ts#L74-L75

and sends out emails

https://github.com/outline/outline/blob/30190866f8c2c6189105d34cb6b508f4917b4564/server/queues/processors/NotificationsProcessor.ts#L125

which is still the case https://github.com/outline/outline/blob/11b823e24266201b2c61a227d3d38c7dd4d2295e/server/queues/processors/NotificationsProcessor.ts#L167 (on **this branch**) although now we also include subscriptions 

https://github.com/outline/outline/blob/11b823e24266201b2c61a227d3d38c7dd4d2295e/server/queues/processors/NotificationsProcessor.ts#L139

I think this https://github.com/outline/outline/blob/11b823e24266201b2c61a227d3d38c7dd4d2295e/server/queues/processors/NotificationsProcessor.test.ts#L211 and https://github.com/outline/outline/blob/11b823e24266201b2c61a227d3d38c7dd4d2295e/server/queues/processors/NotificationsProcessor.test.ts#L255 test specifically https://github.com/outline/outline/blob/11b823e24266201b2c61a227d3d38c7dd4d2295e/server/queues/processors/NotificationsProcessor.test.ts#L309-L311 should prove the functionality.

Let me know if this sounds alright.

Reply:
I don't really follow, but we can move the check into the beforeSend, like here:

https://github.com/outline/outline/blob/863f22750ffb86fec628901a8380d285e5c12ab3/server/emails/templates/InviteAcceptedEmail.tsx#L27

Reply:
[abfc28f](https://github.com/outline/outline/pull/3834/commits/abfc28f83e99f08a9bc7f733f69a8c6f75a4d3d4) commit should better illustrate it. 

I fear we might flood user inboxes because both `revisions.create` and `documents.update` event will send emails based on same source event.

Assumption based on 
``` js
async documentUpdated(event: DocumentEvent | RevisionEvent)
```
and
https://github.com/outline/outline/blob/086c3ec2d83d6603f8834b0a2ef5229f9638bef4/server/queues/processors/NotificationsProcessor.ts#L62-L65